morango/models.py

```diff
@@ -113,20 +122,33 @@ class SyncSession(models.Model):
     the current transfer happening for this sync session.
     """
 
-    id = models.UUIDField(primary_key=True)
+    id = UUIDField(primary_key=True)
 
     # track when the session started and the last time there was activity for this session
     start_timestamp = models.DateTimeField(default=timezone.now)
     last_activity_timestamp = models.DateTimeField(blank=True)
+    active = models.BooleanField(default=True)
+
+    # track whether this device is acting as the server for the sync session
+    is_server = models.BooleanField(default=False)
 
     # track the certificates being used by each side for this session
     local_certificate = models.ForeignKey(Certificate, blank=True, null=True, related_name="syncsessions_local")
     remote_certificate = models.ForeignKey(Certificate, blank=True, null=True, related_name="syncsessions_remote")
 
+    # track the morango profile this sync session is happening for
+    profile = models.CharField(max_length=40)
+
     # information about the connection over which this sync session is happening
     connection_kind = models.CharField(max_length=10, choices=[(u"network", u"Network"), (u"disk", u"Disk")])
-    connection_path = models.CharField(max_length=1000)  # file path if kind=disk, and base URL if kind=network
-    connection_params = models.TextField(default=u"{}")
+    connection_path = models.CharField(max_length=1000)  # file path if kind=disk, and base URL of server if kind=network
+
+    # for network connections, keep track of the IPs on either end
+    local_ip = models.CharField(max_length=100, blank=True)
+    remote_ip = models.CharField(max_length=100, blank=True)
+
+    local_instance = models.TextField(default=u"{}")
+    remote_instance = models.TextField(default=u"{}")
```

Comment:
This is the full serialized instance (all the input params), for tracking/debugging purposes.

The instance won't change without a server restart, which in general shouldn't happen in the midst of a sync, but in principle could (especially on the server side). But since the only reason we track it here is to support debugging/tracking, it wouldn't cause problems if it changed.